File: 2024-2/0682/unidad07.f - Programación distribuida - reportes/SoftPub_Front-end/SoftPubPersonaBO/SeccionBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubBaseBO;
using SoftPubBaseBO.ServicioWeb;

namespace SoftPubPersonaBO
{
    public class SeccionBO: BaseBO
    {  
        public int insertar(string idSeccion, string nombre_seccion, string departamento)
        {
            return this.WsCliente.seccion_insertar(idSeccion, nombre_seccion, departamento);
        }

        public int modificar(string idSeccion, string nombre_seccion, string departamento)
        {
            return this.WsCliente.seccion_modificar(idSeccion, nombre_seccion, departamento);
        }

        public int eliminar(string idSeccion)
        {            
            return this.WsCliente.seccion_eliminar(idSeccion);
        }

        public BindingList<seccion> listarTodos()
        {
            seccion[] arreglo = this.WsCliente.seccion_listarTodos();
            if (arreglo == null)
                return null;
            else
                return new BindingList<seccion>(arreglo);
        }

        public seccion obtenerPorId(string idSeccion)
        {
            return this.WsCliente.seccion_obtenerPorId(idSeccion);
        }
    }
}

File: 2024-2/0682/unidad07.f - Programación distribuida - reportes/ejemplo completo/SoftPub_Back-end/ServidorRMI/SoftPubRMI_Servidor/src/pe/edu/pucp/softpub/rmi/servidor/interfaces/SeccionBO.java
package pe.edu.pucp.softpub.rmi.servidor.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.model.Seccion;

public interface SeccionBO extends Remote{
    
    public Integer insertar(String idSeccion, String nombre_seccion, String departamento) throws RemoteException;
    
    public Integer modificar(String idSeccion, String nombre_seccion, String departamento) throws RemoteException;
    
    public Integer eliminar(String idSeccion) throws RemoteException;
    
    public ArrayList<Seccion> listarTodos() throws RemoteException;
    
    public Seccion obtenerPorId(String idSeccion) throws RemoteException;
}
